### PR TITLE
adds support for workspace select

### PR DESCRIPTION
**Pull request by charleszipp**
This adds support for the first workspace command `select`. This will allow users to switch between existing workspaces within a pipeline. Other workspace commands such as `new`, `delete`, and `show` will be added in future pr's.

### CODE REVIEW

**Comment by favoretti** [comment]
On the phone atm, will have a look in my morning. Great addition!

File: tasks/terraform-cli/task.json [hunk 1]
```diff
@@ -111,7 +112,7 @@
             "label": "Command Options",
             "helpMarkDown": "Any additional options for the selected command such as `-input=true`. Multiple options can be provided delimited by spaces as `-option1=value -option2=value -option3=value`",
             "required": "false",
-            "visibleRule": "command != forceunlock"
+            "visibleRule": "command != forceunlock && command != workspaceselect"
```
Comment (favoretti) [inline tasks/terraform-cli/task.json:115]: Shouldn't this be `&& command != workspace`? Or am I missing concatenation of command and subcommand somewhere?
```suggestion
            "visibleRule": "command != forceunlock && command != workspace"

```

File: tasks/terraform-cli/task.json [hunk 2]
```diff
@@ -252,6 +253,27 @@
...
+            "properties": {
+                "EditableOptions": "False"
+            }
+        },
+        {
+            "name": "workspaceName",
+            "type": "string",
+            "label": "Workspace Name",
+            "required": false,
+            "visibleRule": "command = workspaceselect",
```
Comment (favoretti) [inline tasks/terraform-cli/task.json:275]: Same here, shouldn't this be `workspace`?
```suggestion
            "visibleRule": "command = workspace",

```

File: overview.md [hunk 3]
```diff
@@ -311,3 +312,19 @@ Skip apply if destroy operations
         environmentServiceName: 'My Azure Service Connection'
         commandOptions: '$(System.DefaultWorkingDirectory)/terraform.tfplan'
 ```
+
+## **Workspaces (NEW)**
+
+The task supports managing workspaces within a pipelines. The following workspace subcommands are supported.
```
Comment (favoretti) [inline overview.md:318]: Nitpicking, but either `within pipelines` or `within a pipeline`?
```suggestion
The task supports managing workspaces within a pipeline. The following workspace subcommands are supported.

```

File: tasks/terraform-cli/src/task.ts [hunk 4]
```diff
@@ -109,4 +110,13 @@ export class Task {
     fmt(): commands.ICommand {
         return new commands.FormatCommandHandler(this.runner);
     }
+
+    workspace(): commands.ICommand {
+        switch(this.ctx.workspaceSubCommand){
+            case "select":
+                return new commands.WorkspaceSelectCommandHandler(this.runner);
+            default:
+                throw new Error(`Workspace sub-command "${this.ctx.workspaceSubCommand}" not supported`);
```
Comment (favoretti) [inline tasks/terraform-cli/src/task.ts:119]: Also nit-picking :) `is not supported.`?
```suggestion
                throw new Error(`Workspace sub-command "${this.ctx.workspaceSubCommand}" is not supported`);

```

**Review by favoretti — COMMENTED** [review]
Left a few comments, but otherwise LGTM.

File: overview.md (hunk 3 above)
Comment (charleszipp) [reply overview.md]: good catch. fixed!

File: tasks/terraform-cli/src/task.ts (hunk 4 above)
Comment (charleszipp) [reply tasks/terraform-cli/src/task.ts]: another good catch. fixed!

File: tasks/terraform-cli/task.json (hunk 2 above)
Comment (charleszipp) [reply tasks/terraform-cli/task.json]: fixed. not necessarily workspace but, the workspace subcommand. `workspace list` for example doesnt take a name parameter. So the rule is based on the subcommand.

File: tasks/terraform-cli/task.json (hunk 1 above)
Comment (charleszipp) [reply tasks/terraform-cli/task.json]: fixed. not necessarily workspace but, the workspace subcommand. `workspace select` for example doesnt take any options. So the rule is based on the subcommand.

**Review by favoretti — APPROVED** [review]

**Comment by Radamonas** [comment]
Its a breaking change.

**Comment by favoretti** [comment]
@Radamonas How come?

**Comment by favoretti** [comment]
Ah, I see. Thanks @charleszipp!

**Pull request merged**